### PR TITLE
Make transaction status stream work

### PR DESCRIPTION
**Pull request by freesig**
This PR refactors the `transaction_status_change` function to improve its readability and efficiency. The main changes are:

1. Changed input stream type from `BoxStream<'a, Result<TxUpdate, BroadcastStreamRecvError>>` to `BoxStream<'a, TxStatusMessage>`.
2. Changed return type from `impl Stream<Item = anyhow::Result<TransactionStatus>> + 'a` to `impl Stream<Item = anyhow::Result<ApiTxStatus>> + 'a`.
3. Removed unnecessary async/await and error mapping for the initial database status check.
4. Replaced the complex unfold logic with a simpler chain of the initial database check and stream, followed by a take_until to handle stream termination.
5. Replaced the nested match structure with a single match block to improve readability.
6. Simplified the stream termination logic by closing the stream if s new transaction status is not Submitted.

Additionally, this PR introduces property testing to ensure the refactored `transaction_status_change` function behaves correctly under various input conditions. This will help identify potential edge cases, increase the test coverage, and provide greater confidence in the robustness of the changes made.

### CODE REVIEW

**Comment by Voxelot** [comment]
We'll likely also want to consider this issue as well, because without a terminal message indicating the closing of the stream it makes error handling on the client side more difficult: https://github.com/FuelLabs/fuel-core/issues/823

File: crates/fuel-core/src/service/query.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+use crate::{
+    query::transaction_status_change,
+    schema::tx::types::TransactionStatus,
+};
+
+use super::*;
+
+impl FuelService {
+    /// Submit a transaction to the txpool.
+    pub fn submit(&self, mut tx: Transaction) -> anyhow::Result<InsertionResult> {
```
Comment (Voxelot) [inline crates/fuel-core/src/service/query.rs:28]: 👍🏻 Putting the rust-level APIs here will be helpful for migrating to other RPCs and also for those using fuel-core as a library (indexer)

File: crates/services/txpool/src/service/update_sender.rs [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        C: CreateChannel,
+    {
+        // Remove closed senders from the list.
+        // Careful not to hold the lock while awaiting.
+        {
+            remove_closed(&mut self.senders.lock());
+        }
+
+        // Acquire a permit from the semaphore asynchronously.
+        let permit = Arc::clone(&self.permits).acquire().await;
```
Comment (Voxelot) [inline crates/services/txpool/src/service/update_sender.rs:244]: Could this cause messages to get missed if the creation of the sender is awaiting a permit?

Comment (mitch-fuel) [reply crates/services/txpool/src/service/update_sender.rs:244]: are you suggesting a message wouldn't be sent in this scenario?

Comment (mitch-fuel) [reply crates/services/txpool/src/service/update_sender.rs:244]: or that the transaction could complete before it is possible to finalize the subscription?

Comment (mitch-fuel) [reply crates/services/txpool/src/service/update_sender.rs:244]: we could require subscribing before the transaction is submitted given that the id is known but this could be abused. another option would be holding an LRU of recent tx that can still be subscribed for, but that supposes timing requirements

Comment (freesig) [reply crates/services/txpool/src/service/update_sender.rs:244]: Yes it will cause messages to be missed but we can't garauntee any sort of message delivery anyway

**Review by xgreenx — APPROVED** [review]
@freesig  I made small changes around the main logic. Could you check it, please? If everything okay, I think we can merge it=)

File: crates/fuel-core/src/service/adapters/txpool.rs [hunk 3]
```diff
@@ -147,7 +148,10 @@ impl fuel_core_txpool::ports::TxPoolDb for Database {
     fn transaction_status(
         &self,
         tx_id: &fuel_core_types::fuel_types::Bytes32,
-    ) -> StorageResult<Option<fuel_core_types::services::txpool::TransactionStatus>> {
-        Ok(self.get_tx_status(tx_id)?)
+    ) -> StorageResult<fuel_core_types::services::txpool::TransactionStatus> {
```
Comment (freesig) [inline crates/fuel-core/src/service/adapters/txpool.rs:151]: Kinda weird to return an error for not found

**Review by freesig — COMMENTED** [review]
looks good
